eos-update-flatpak-repos

```diff
@@ -1476,6 +1503,9 @@ def _migrate_installed_flatpaks():
                                                     arch, new_branch)
                         refs.append(new_ref)
 
+                        if migrate_extra:
+                            migrate_extra_data(ref, new_ref)
```

Comment:
What will happen if the system loses power mid-way through this step? I think what will happen is that, on next boot, we'll enter the "Found already installed" branch above, the old copy of Skype will be uninstalled, and the user will be left without a working Skype app until they upgrade it.

I think we need to run this step on both the `new_ref` and `not new_ref` branches, and make `migrate_extra_data` do something like:

```

    if os.path.exists(new_extra):
        # either the user manually installed Skype from Flathub,
        # or we got this far but died before uninstalling the old ref last time
        logging.info('%s already has extra data dir', new_ref)
    else:
        logging.debug('Copying extra data from {} to {}'.format(old_ref, new_ref))
        subprocess.check_call(['cp', '-al', old_extra, new_extra + '.tmp'])
        os.rename(new_extra + '.tmp', new_extra)
```